File: Algorithms/ReactNativeAlgorithms/src/algorithms/selectSort.js
export default function selectSort(list) {
  for(var i = 0; i < list.length - 1; i++) {
    var min_el_pos = i;
    for(var j = min_el_pos + 1; j < list.length; j++) {
      if(list[j] < list[min_el_pos]) {
        min_el_pos = j;
      }
    }
    // swap
    var temp = list[i];
    list[i] = list[min_el_pos];
    list[min_el_pos] = temp;
  }
  return list;
}

File: Algorithms/Swift Algorithms/Swift Algorithms/Algorithms/SelectSort.swift
//
//  SelectSort.swift
//  Swift Algorithms
//
//  Created by Krzysztof Drab on 17/05/2020.
//  Copyright © 2020 Krzysztof Drab. All rights reserved.
//

import Foundation

func selectSort(list: inout [Double]) -> [Double] {
    for i in 0..<list.count {
        var minElPos = i;
//        var minEl = list[minElPos];
        for j in (minElPos + 1)..<list.count {
            if list[j] < list[minElPos] {
                minElPos = j;
            }
//            if list[j] < minEl {
//                minElPos = j;
//                minEl = list[minElPos];
//            }
        }
        // swap
//        let temp = list[i];
//        list[i] = list[minElPos];
//        list[minElPos] = temp;
        list.swapAt(i, minElPos);
    }
    return list;
}
